gcloud-java-compute/src/test/java/com/google/gcloud/compute/SnapshotTest.java

```diff
@@ -0,0 +1,235 @@
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.collect.ImmutableList;
+
+import org.junit.Test;
+
+import java.util.List;
+
+public class SnapshotTest {
+
+  private static final String ID = "42";
+  private static final DiskId SOURCE_DISK = DiskId.of("project", "zone", "disk");
+  private static final Long CREATION_TIMESTAMP = 1453293540000L;
+  private static final String DESCRIPTION = "description";
+  private static final List<LicenseId> LICENSES = ImmutableList.of(
+      LicenseId.of("project", "license1"), LicenseId.of("project", "license2"));
+  private static final SnapshotId SNAPSHOT_ID = SnapshotId.of("project", "snapshot");
+  private static final SnapshotInfo.Status STATUS = SnapshotInfo.Status.CREATING;
+  private static final Long DISK_SIZE_GB = 42L;
+  private static final String SOURCE_DISK_ID = "diskId";
+  private static final Long STORAGE_BYTES = 24L;
+  private static final SnapshotInfo.StorageBytesStatus STORAGE_BYTES_STATUS =
+      SnapshotInfo.StorageBytesStatus.UP_TO_DATE;
+
+  private Compute serviceMockReturnsOptions = createStrictMock(Compute.class);
+  private ComputeOptions mockOptions = createMock(ComputeOptions.class);
+  private Compute compute;
+  private Snapshot snapshot;
+  private Snapshot expectedSnapshot;
+
+  private void initializeExpectedSnapshot(int optionsCalls) {
+    expect(serviceMockReturnsOptions.options()).andReturn(mockOptions).times(optionsCalls);
+    replay(serviceMockReturnsOptions);
+    expectedSnapshot = new Snapshot.Builder(serviceMockReturnsOptions, SNAPSHOT_ID, SOURCE_DISK)
+        .id(ID)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .description(DESCRIPTION)
+        .status(STATUS)
+        .diskSizeGb(DISK_SIZE_GB)
+        .licenses(LICENSES)
+        .sourceDiskId(SOURCE_DISK_ID)
+        .storageBytes(STORAGE_BYTES)
+        .storageBytesStatus(STORAGE_BYTES_STATUS)
+        .build();
+    compute = createStrictMock(Compute.class);
+  }
+
+  private void initializeSnapshot() {
+    snapshot = new Snapshot.Builder(compute, SNAPSHOT_ID, SOURCE_DISK)
+        .id(ID)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .description(DESCRIPTION)
+        .status(STATUS)
+        .diskSizeGb(DISK_SIZE_GB)
+        .licenses(LICENSES)
+        .sourceDiskId(SOURCE_DISK_ID)
+        .storageBytes(STORAGE_BYTES)
+        .storageBytesStatus(STORAGE_BYTES_STATUS)
+        .build();
+  }
+
+  @Test
+  public void testToBuilder() {
+    initializeExpectedSnapshot(8);
+    compareSnapshot(expectedSnapshot, expectedSnapshot.toBuilder().build());
+    Snapshot newSnapshot = expectedSnapshot.toBuilder().description("newDescription").build();
```

Comment:
Nit: could also set the snapshotId and sourceDisk (those builder methods aren't covered by tests yet)